testing/src/main/java/org/oppia/android/testing/AssertionHelpers.kt

```diff
@@ -4,9 +4,18 @@ import org.junit.Assert.fail
 import kotlin.reflect.KClass
 import kotlin.reflect.full.cast
 
+/* This file contains generic assertion helpers which are meant to be usable in any test.
+   Not for methods which are specific to a certain test or part of the codebase. */
+
 /**
- * Assertion helpers which are generic methods to be used at multiple places
- * Not for methods which are specific to a certain test or part of the codebase
+ * A replacement to JUnit5's assertThrows().
+ *
+ * Takes a parameter 'type' which defines the expected exception type and another parameter
+ * 'operation' which defines the function to be performed.
+ *
+ * Asserts that execution of the supplied operation throws an exception of the supplied type.
+ *
+ * Returns the exception.
```

Comment:
Awesome documentation comment! Just one suggested change: when describing parameters & return values, you should either combine this in a brief one-line comment, or split it and use KDoc tags to specifically document each piece. The two versions based on this situation might look like:

```kotlin
/** Asserts the specified operation throws an exception of the specified type and returns the exception. */
```

(needs line-wrapping)

vs:

```kotlin
/**
 * A replacement to JUnit5's assertThrows() that asserts an execution of the ...
 * An example of this might be:
 *  ... add pre-formatted example
 *
 * @param type the type of exception to be thrown
 * @param operation the operation being run
 * @return the exception being thrown
 * @throws AssertionError if the specified exception is not thrown
 */
```

Notice that one is much higher-level and the latter is more structured. In most cases, you can choose either style based on your preference. In some cases (e.g. especially complex contracts or long parameter lists) we prefer the latter style.

Please update the documentation accordingly.

Reply:
Updated accordingly. Please have a look.